pkg/sql/opt/optbuilder/scalar.go

```diff
@@ -500,6 +501,21 @@ func (b *Builder) buildFunction(
 		return b.finishBuildGeneratorFunction(f, out, inScope, outScope, outCol)
 	}
 
+	// Add a dependency on sequences that are used as a string argument.
+	if b.trackViewDeps {
+		seq, found, err := sqlbase.GetSequenceFromFunc(f)
+		if err != nil {
+			panic(err)
+		}
+		if found {
+			tn := tree.MakeUnqualifiedTableName(tree.Name(seq))
+			ds, _ := b.resolveDataSource(&tn, privilege.SELECT)
```

Comment:
What if resolve datasource returns an error? I think you only want to ignore it if it is a "UnknownDescriptor" error (or something like that)

Reply:
This should be okay because resolveDataSource will panic if theres an error, as Radu mentioned the 2nd return value is the datasource name and not error.